### DISCUSSION TITLE
Command-line selecting task and an Alfred workflow!

### DISCUSSION THREAD

**Discussion by flightmansam** [minbrowser/min#2008] [Show and tell]
## Background
For those of you who use [Alfred](https://www.alfredapp.com/) or the built-in Spotlight on macOS, you might have become quite used to bashing away at finding your files or doing search queries with a simple ⌘+space and search. E.g.

[![E.g.](https://img.youtube.com/vi/ite5EiccGgI/0.jpg)](https://youtu.be/ite5EiccGgI "Alfred (YouTube)")

I have really really enjoyed optimising my different research interests into different tasks and often I will be using an Alfred search to quickly search a new tab when I am away from the browser (e.g. googling stackexchange!!). However, since any search query will go straight into the currently active task there is a risk that when I do an Alfred search it starts to make my nicely organised tasks messy with other searches. 

So I had a go at adding a new command line argument for min with a `-t <taskname-query>` format. This can be tried out on my fork under the branch https://github.com/flightmansam/min/tree/task-cmd-line-args. So now if you supply a query in the -t field it will try to find that task (and if not, create it!). You can also pass in just the task id so if you already know exactly where to go. 

I then paired it with my custom [Alfred workflow](https://github.com/flightmansam/min_alfred_workflow) that will give the following result (NB: I am using my personal fork of min so the UI is a little different):

[![E.g.](https://img.youtube.com/vi/WeVtnPtgGwI/0.jpg)](https://youtu.be/WeVtnPtgGwI "Alfred workflow with  new Min command line API (YouTube)")

Pretty happy with my efforts today. Any suggestions?

For Linux users there is no reason someone couldn't make an [Albert](https://github.com/albertlauncher/albert) extension (or maybe gnome search?).

For windows users maybe someone can write an extension for [PowerToys Run](https://docs.microsoft.com/en-us/windows/powertoys/run) or [Wox](https://github.com/Wox-launcher/Wox).

**Comment by PalmerAL** [comment]
This seems like a reasonable thing to add, do you want to make a PR?
